crates/ra_project_model/src/lib.rs

```diff
@@ -25,15 +25,35 @@ pub use crate::{
 };
 
 #[derive(Clone, PartialEq, Eq, Hash, Debug)]
-pub struct CargoTomlNotFoundError(pub PathBuf);
+pub struct CargoTomlNoneFoundError(pub PathBuf);
 
-impl std::fmt::Display for CargoTomlNotFoundError {
+#[derive(Clone, PartialEq, Eq, Hash, Debug)]
+pub struct CargoTomlMultipleValidFoundError(pub Vec<PathBuf>);
+
+#[derive(Clone, PartialEq, Eq, Hash, Debug)]
+pub struct CargoTomlSearchFileSystemError(pub PathBuf, pub String);
```

Comment:
I went this route to enumerate the possibilities, could also have been one error like:

```
pub enum CargoTomlFindErrorType {
    NoneFound
    MultipleValidFound
    FilesystemError
}

CargoTomlFindError(pub PathBuf, typ CargoTomlFindErrorType)
```

However I don't this is "rusty" or a common practice in general nor in this codebase?

Reply:
Let's stick with a single error type here

```rust
struct CargoTomlNotFoundError {
    searched_at: PathBuf,
    reason: String,
}
```